src/main/java/org/testng/TestNG.java

```diff
@@ -1394,7 +1394,9 @@ private SuiteRunner createSuiteRunner(XmlSuite xmlSuite) {
     }
 
     for (IReporter r : result.getReporters()) {
-      addListener(r);
+      if (!m_reporters.containsKey(r.getClass())) {
```

Comment:
Yes, but we can add a new boolean param in order to determine the behavior.

In fact, the question is more: do we need to add all listener implementations? If yes, we should not check only with `m_reporters`. If no, we should add the listener directly in `m_reporters`.
WDYT about it?